The patch below fixes a spurious call to mach_port_deallocate: rootdevid
is already deallocated as soon as it is fetched through __io_identity.

---
 sysdeps/mach/hurd/getcwd.c |    1 -
 1 file changed, 1 deletion(-)

--- a/sysdeps/mach/hurd/getcwd.c
+++ b/sysdeps/mach/hurd/getcwd.c
@@ -60,7 +60,6 @@
       __mach_port_deallocate (__mach_task_self (), thisid);
       __mach_port_deallocate (__mach_task_self (), thisdevid);
       __mach_port_deallocate (__mach_task_self (), rootid);
-      __mach_port_deallocate (__mach_task_self (), rootdevid);
 
       if (dirbuf != NULL)
 	__vm_deallocate (__mach_task_self (),
